### PR TITLE
[doc] Move each profiler to its own file + Add missing PyTorchProfiler to the doc

### LINKED ISSUE #7806: PytorchProfiler API documentation missing
## 📚 Documentation

In [profilers documentation](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.profiler.profilers.html), there is no documentation for pytorch profilers. I wonder whether the pytorch profiler is the one with `torch.profiler.profile` or the one with `torch.autograd` that is deprecated according to [torch documentation](https://pytorch.org/docs/stable/profiler.html). Besides, how can I generate result files for TensorBoard from the profiler in pytorch lightning？

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #7806

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
imports may brake for some people who do not import at top level from init file

**Review by ethanwharris — APPROVED** [review]

**Comment by tchaton** [comment]
> imports may brake for some people who do not import at top level from init file

Hey @awaelchli. Added a depreciation call.

Best,
T.C

**Review by justusschock — APPROVED** [review]

File: docs/source/api_references.rst [hunk 1]
```diff
@@ -137,8 +137,14 @@ Profiler API
 .. autosummary::
     :toctree: api
     :nosignatures:
+    :template: classtemplate.rst
+
+    AdvancedProfiler
+    BaseProfiler
+    PassThroughProfiler
+    PyTorchProfiler
+    SimpleProfiler
```
Comment (ananthsub) [inline docs/source/api_references.rst:146]: should the abstract profiler also be documented here?

**Comment by Borda** [comment]
Is it rename files or adding new code, from the PR diff it is not clear...

**Review by ethanwharris — APPROVED** [review]

File: docs/source/api_references.rst (hunk 1 above)
Comment (tchaton) [reply docs/source/api_references.rst:146]: Yes. Done !

**Comment by tchaton** [comment]
> Is it rename files or adding new code, from the PR diff it is not clear...

This PR extracts each profiler from profilers.py and move it to simple.py, advanced.py, etc..

Best,
T.C

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Pull request merged**